### PR TITLE
Added root functionality to support enable data on higher versions of android

### PR DESCRIPTION
**Pull request by aljosamrak**
I have added new Preference screen which is visible only on rooted devices. In it, the user can enable to use root commands to enable mobile data and/or GPS.
As far as I know, this is the only way to enable data on newer versions of Android.

I have also added new SMS message to request GPS coordinates by sending an SMS (default: 'gps get'). The device responds with an SMS with GPS coordinates if they are available.

### CODE REVIEW

**Comment by rpellerin** [comment]
Thank your for the PR. I will review it as soon as possible.

**Comment by aljosamrak** [comment]
Take your time. If there are any problems let me know.

Aljoša Mrak

2017-11-16 22:22 GMT+01:00 Romain Pellerin <notifications@github.com>:

File: src/eu/romainpellerin/remotecontrolviasms/CustomArrayAdapter.java [hunk 1]
```diff
@@ -8,11 +8,13 @@
 import android.widget.ImageView;
 import android.widget.TextView;
 
+import java.util.List;
+
 public class CustomArrayAdapter extends ArrayAdapter<String> {
 	private final Context context;
-	private final String[] values;
+	private final List<String> values;
```
Comment (rpellerin) [inline src/eu/romainpellerin/remotecontrolviasms/CustomArrayAdapter.java:15]: Nice improvement, thanks!

File: src/eu/romainpellerin/remotecontrolviasms/activities/MainActivity.java [hunk 2]
```diff
@@ -17,22 +20,29 @@
...
 import eu.romainpellerin.remotecontrolviasms.CustomArrayAdapter;
 import eu.romainpellerin.remotecontrolviasms.MyPreferenceFragment;
 import eu.romainpellerin.remotecontrolviasms.PowerButtonService;
 import eu.romainpellerin.remotecontrolviasms.R;
+import eu.romainpellerin.remotecontrolviasms.RootUtils;
 
 public class MainActivity extends Activity {
 
-	private String[] menuDrawer; // va chercher les strings dans values/strings à afficher dans le menu
+    private ArrayList<String> menuDrawer; // va chercher les strings dans values/strings à afficher dans le menu
```
Comment (rpellerin) [inline src/eu/romainpellerin/remotecontrolviasms/activities/MainActivity.java:39]: Good job!

**Comment by rpellerin** [comment]
I reviewed the code, good job, thanks!
I however did not test it since Android Studio is not installed on my laptop and I was too lazy to do it. I believe you tested your code and it works :) Therefore I'm merging your PR :) Thanks again!

**Comment by aljosamrak** [comment]
Thanks.
If any problems occur in the future please let me know.
I hope you will also update the app on Google Play.
Thanks for writing and publishing the app :).

**Comment by rpellerin** [comment]
Sure thing! I'd love to update the app on Google play however it seems that importing the app in Android Studio requires tweaking the dependencies a bit. Mind doing a PR if you develop on Android Studio to save me the hassle?

**Comment by aljosamrak** [comment]
Sure, I'll import it. No problem.

**Pull request merged**